File: Leetcode/797.cpp
constexpr int MAXN = 15;
class Solution {
public:
    vector<vector<int>> ans;
    int n;
    void dfs(vector<int> v, const vector<vector<int>> &AL) {
        if (v.back() == n-1) {
            ans.emplace_back(v);
            return;
        }
        for (const auto &j : AL[v.back()]) {
            if (count(v.begin(),v.end(),j) == 0) {
                v.emplace_back(j);
                dfs(v, AL);
                v.pop_back();
            }
        }
    }
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        decltype(ans)().swap(ans);
        n = (int)graph.size();
        vector<int> v;
        v.emplace_back(0);
        dfs(v, graph);
        return ans;
    }
};

File: Leetcode/797.go
func count(s []int, elem int) int {
	ans := 0
	for _, i := range s {
		if i == elem {
			ans++
		}
	}
	return ans
}
func dfs(v []int, graph *[][]int, ans *[][]int) {
	if v[len(v)-1] == len(*graph)-1 {
		*ans = append(*ans, v)
		return
	}
	for _, i := range (*graph)[v[len(v)-1]] {
		if count(v, i) == 0 {
			temp := make([]int, len(v))
            copy(temp, v)
			temp = append(temp, i)
			dfs(temp, graph, ans)
		}
	}
}
func allPathsSourceTarget(graph [][]int) [][]int {
	var v []int
	var ans [][]int
    v = append(v, 0)
	dfs(v, &graph, &ans)
	return ans
}